src/encoded/static/components/matrix_human_donor.js

```diff
@@ -0,0 +1,744 @@
+import { tintColor, isLight } from './datacolors';
+import * as globals from './globals';
+import { MATRIX_VISUALIZE_LIMIT } from './matrix';
+import { FacetList, SearchControls } from './search';
+import { BodyMapThumbnailAndModal } from './body_map';
+
+
+/** General collection of attributes to exclude in some scenarios */
+const excludedAttributes = ['', 'n/a', 'unknown', undefined, null];
+
+/** life stages to display on UI */
+const includedLifeStage = ['child', 'embryonic', 'newborn'];
+
+/** Roman mythhology equivalent represening gender */
+const getGenderMythology = (genderType) => {
+    if (genderType === 'male') {
+        return 'mars';
+    }
+
+    if (genderType === 'female') {
+        return 'venus';
+    }
+
+    return '';
+};
+
+/** Query keys to remove from context's urls */
+const removedQueryKeys = ['@id', 'search_base'];
+
+/**
+ *  Key/value pairs mapping dseases to css classes
+ */
+const diseaseCode = {
+    'multiple sclerosis': 'multiple-sclerosis-symbol',
+    'Alzheimer\'s disease': 'alzheimers-disease-symbol',
+    'mild cognitive impairment': 'mild-cognitive-impairment-symbol',
+    'Cognitive impairment': 'cognitive-impairment-symbol',
+    'amyotrophic lateral sclerosis': 'bamyotrophic-lateral-sclerosis-symbol',
+    'nonobstructive coronary artery disease': 'nonobstructive-coronary-artery-symbol',
+    'squamous cell carcinoma': 'squamous-cell-carcinoma-symbol',
+    'basal cell carcinoma': 'basal-cell-carcinoma-symbol',
+};
+
+/**
+ * Gets data like age and gender and adds them to the Matrix
+ *
+ * @param {object} data from a fetch call
+ */
+const addExtraRowHeaderDataToMatrix = (data, matrixUpdateCallBack) => {
+    const graph = data['@graph'].map((g) => {
+        let age = '';
+
+        if (g.age) {
+            const ageText = `${[g.age].filter((a) => !excludedAttributes.includes(a)).join(', ')}`;
+            const formattedAge = `${(ageText || '').replace(' or above', '+')} ${ageText ? g.age_units : ''}${ageText === 1 || ageText === '' ? '' : 's'}`.trim('');
+            const lifeStageText = g.lifeStage ? ` (${g.lifeStage})` : '';
+
+            age = `${formattedAge}${lifeStageText}`;
+        }
+
+        return {
+            accession: g['@id']?.replace('/human-donors/', '')?.replace('/', ''),
+            age,
+            genderMythology: getGenderMythology(g.sex),
+            genderType: g.sex,
+            lifeStage: includedLifeStage.includes(g.life_stage) ? g.life_stage : '',
+        };
+    });
+
+    matrixUpdateCallBack(graph);
+};
+
+/**
+ *  Matrix data from context, to be used to paint the DOM
+ *
+ * @param {context} context
+ */
+const getDataTableData = (context) => {
+    const { matrix } = context;
+    const x = {
+        header: matrix.x.assay_title.buckets.map((assayTitle) => assayTitle.key),
+    };
+    const xLength = x.header.length;
+    const y = [];
+
+    matrix.y['replicates.library.biosample.donor.accession'].buckets.forEach((accessionBucket) => {
```

Comment:
Technically you should retrieve `replicates.library.biosample.donor.accession` from matrix.y.group_by[0] so that this code automatically adjusts if you change experiment.py. At this stage, though, this seems unlikely to change.

Reply:
Are you okay with me not changing this? If there is a change to experiment.py. the code needs to be updated to reflect it and being explicit like using "replicates.library.biosample.donor.accession", to me, forces me to remember that.